src/Tooltip.js

```diff
@@ -156,6 +168,11 @@ class Tooltip extends React.Component {
   addTargetEvents() {
     this._target.addEventListener('mouseover', this.onMouseOverTooltip, true);
     this._target.addEventListener('mouseout', this.onMouseLeaveTooltip, true);
+    this._target.addEventListener('keydown', this.onEscKeyDown, true);
+    this._target.addEventListener('focusin', this.handleFocus, true);
```

Comment:
Because this calls the same method as `focusout` which just blindly toggles the tooltip, it can get into a bad state where `focusin` hides the toggle and `focusout` shows it. We probably want to explicitly show on `focusin` and hide on `focusout`
![tooltip-focus](https://user-images.githubusercontent.com/664714/40186050-268b576e-59c2-11e8-864e-6f4b70e4188f.gif)
 
Also, this is more evident when using `esc` to close the tooltip and then tabbing away.
![tooltip-esc](https://user-images.githubusercontent.com/664714/40186433-13325c3e-59c3-11e8-927c-581490b85b6a.gif)